src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java

```diff
@@ -0,0 +1,297 @@
+     * altering an inventories contents. Any implementers who are able to detect
+     * modification through this method should throw an exception.
+     * <p>
+     * SERIOUSLY: DO NOT MODIFY THE RETURNED ITEMSTACK
+     *
+     * @param slot Slot to query
+     * @return ItemStack in given slot. May be null.
+     **/
+    @Override
+    public ItemStack getStackInSlot(int slot)
+    {
+        int activeSlot = slot;
+
+        for (final IItemHandlerModifiable modifiable : handlers)
+        {
+            if (slot < modifiable.getSlots())
+            {
+                return modifiable.getStackInSlot(activeSlot);
+            }
+
+            activeSlot -= modifiable.getSlots();
+        }
+
+        return null;
+    }
+
+    /**
+     * Inserts an ItemStack into the given slot and return the remainder.
+     * The ItemStack should not be modified in this function!
+     * Note: This behaviour is subtly different from IFluidHandlers.fill()
+     *
+     * @param slot     Slot to insert into.
+     * @param stack    ItemStack to insert.
+     * @param simulate If true, the insertion is only simulated
+     * @return The remaining ItemStack that was not inserted (if the entire
+     * stack is accepted, then return null). May be the same as the input
+     * ItemStack if unchanged, otherwise a new ItemStack.
+     **/
+    @Override
+    public ItemStack insertItem(int slot, final ItemStack stack, final boolean simulate)
+    {
+        int activeSlot = slot;
+
+        for (final IItemHandlerModifiable modifiable : handlers)
+        {
+            if (activeSlot < modifiable.getSlots())
+            {
+                return modifiable.insertItem(activeSlot, stack, simulate);
+            }
+
+            activeSlot -= modifiable.getSlots();
+        }
+
+        return stack;
+    }
+
+    /**
+     * Extracts an ItemStack from the given slot. The returned value must be
+     * null if nothing is extracted, otherwise it's stack size must not be
+     * greater than amount or the itemstacks getMaxStackSize().
+     *
+     * @param slot     Slot to extract from.
+     * @param amount   Amount to extract (may be greater than the current stacks
+     *                 max limit)
+     * @param simulate If true, the extraction is only simulated
+     * @return ItemStack extracted from the slot, must be null, if nothing can
+     * be extracted
+     **/
+    @Override
+    public ItemStack extractItem(int slot, final int amount, final boolean simulate)
+    {
+        int activeSlot = slot;
+
+        for (final IItemHandlerModifiable modifiable : handlers)
+        {
+            if (activeSlot < modifiable.getSlots())
+            {
+                return modifiable.extractItem(slot, amount, simulate);
+            }
+
+            activeSlot -= modifiable.getSlots();
+        }
+
+        return null;
+    }
+
+    protected IItemHandlerModifiable[] getHandlers()
+    {
+        return handlers;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Return a copy of "handlers". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)